frontend/src/pages/dashboard/components/table/drivers-table/driver-table.tsx

```diff
@@ -0,0 +1,78 @@
+import { type DriverCreateUpdateRequestDto } from 'shared/build/index.js';
+
+import { Button, Modal, Table } from '~/libs/components/components.js';
+import { DataStatus } from '~/libs/enums/data-status.enum.js';
+import {
+  useAppDispatch,
+  useAppSelector,
+  useAppTable,
+  useCallback,
+  useState,
+} from '~/libs/hooks/hooks.js';
+import { type DriverGetAllResponseDto } from '~/libs/types/types.js';
+import { AddDriverForm } from '~/pages/business/components/add-driver-form/add-driver-form.js';
+import { getDriversPage } from '~/slices/driver-table/actions.js';
+import { actions } from '~/slices/driver-table/driver-table.js';
+
+import { columns } from './columns/columns.js';
+import styles from './styles.module.scss';
+
+const DriverTable: React.FC = () => {
+  const dispatch = useAppDispatch();
+
+  const data = useAppSelector((state) => state.driversTable.drivers);
+  const total = useAppSelector((state) => state.driversTable.total);
+  const status = useAppSelector((state) => state.driversTable.dataStatus);
+
+  const [isActiveModal, setIsActiveModal] = useState(false);
+
+  const tableHook = useAppTable<DriverGetAllResponseDto>({
+    tableFetchCall: getDriversPage,
+  });
+
+  const handleOpenModal = useCallback(() => {
+    setIsActiveModal(true);
+  }, []);
+
+  const handleCloseModal = useCallback(() => {
+    setIsActiveModal(false);
+  }, []);
+
+  const handleSubmit = useCallback(
+    (payload: DriverCreateUpdateRequestDto) => {
+      void dispatch(actions.addDriver({ payload })).then(tableHook.updatePage);
```

Comment:
table should re-render when slice changes, if you need to make additional api call - do this in actions (dispatch another action 🤔 )

Reply:
For example, if we stay on the first page and add a new driver we get this driver just after reloading. To update the slice we should call the current page but pageIndex and pageSize we have just inside our hook, that’s why we can't do this in actions

Reply:
during adding a new driver we don't need to know the current page. new added driver will be always on the 1st page at the top. so refactor it as Pasha suggests

Reply:
We need the current page and size not for adding a new driver (it will be always first by sorting), we need it to update the current table page and show this driver without reloading

Reply:
You can sync all options via query params.